IBEngine/IBRendererFrontend.cpp

```diff
@@ -17,7 +17,25 @@ namespace IB
         fromBinary(stream, &mesh->VertexCount);
         fromBinary(stream, &mesh->IndexCount);
 
-        mesh->Vertices = fromBinary<Vertex>(stream, mesh->VertexCount * sizeof(Vertex));
-        mesh->Indices = fromBinary<uint16_t>(stream, mesh->IndexCount * sizeof(uint16_t));
+        mesh->Vertices = fromBinary<Vertex*>(stream, mesh->VertexCount * sizeof(Vertex));
+        mesh->Indices = fromBinary<uint16_t*>(stream, mesh->IndexCount * sizeof(uint16_t));
+    }
+
+    void toBinary(FileStream *stream, ShaderAsset const &shaders)
+    {
+        toBinary(stream, shaders.VertexShaderSize);
```

Comment:
Maybe it is just me, but this is confusing that both these methods have the same name.  Could be because I am only looking at the diff without much extra context but it I originally thought this was calling the function recursively but that didn't make sense due to the parameters being passed.  Not sure of best solution but maybe a serialization namespace so we are doing Serialization::toBinary?

Reply:
That's a good idea - I'll do that!